addons/stock/models/stock_move.py

```diff
@@ -1230,8 +1247,8 @@ def _recompute_state(self):
                 else:
                     move.state = 'confirmed'
 
-    def _get_upstream_documents_and_responsibles(self, visited):
-        if self.move_orig_ids and any(m.state not in ('done', 'cancel') for m in self.move_orig_ids):
+    def _get_upstream_documents_and_responsibles(self, visited, lazy=False):
```

Comment:
maybe it's time to add a proper docstring
```
        """_get_upstream_documents_and_responsibles

        :param visited:
        :param lazy: (Default value = False)
        :returns:
        :rtype: tuple
        """
```

Reply:
maybe rename _get_upstream_documents_and_responsibles to _get_documents_and_reponsibles and add a kwarg direction='upstream' and pass direction=False when you don't want to move thorugh the chain?

@amoyaux